### PR TITLE
Allow skipping unnamed blobs when pushing to file store

### LINKED ISSUE #213: Allow skipping unnamed blobs when pushing to file store
Currently, when an unnamed blob is pushed to a file store, it will be stored in a memory fallback storage by default (See https://github.com/oras-project/oras-go/blob/main/content/file/file.go#L213-L215). 
However, in some cases, clients may not want such blobs to be stored at all, even in the memory.
We should provide an option in the [file store](https://github.com/oras-project/oras-go/blob/main/content/file/file.go#L75) to allow skipping unnamed blobs.

### PR DESCRIPTION
**Pull request by m5i-work**
Resolves #213 

Signed-off-by: Wei Meng <wemeng@microsoft.com>

### CODE REVIEW

File: content/file/file.go [hunk 1]
```diff
@@ -191,26 +196,34 @@ func (s *Store) Fetch(ctx context.Context, target ocispec.Descriptor) (io.ReadCl
...
 		return err
 	}
 
 	return s.graph.Index(ctx, s, expected)
 }
 
 // push pushes the content, matching the expected descriptor.
-// If name is not specified in the descriptor,
-// the content will be pushed to the fallback storage.
+// If name is not specified in the descriptor, the content will be either
```
Comment (Wwwsylvia) [inline content/file/file.go:218]: I think it'd be better to call out the default behavior first. Something like:
> the content will be pushed to the fallback storage by default, or will be discarded when Store.IgnoreNoName is set.

File: content/file/file_test.go [hunk 2]
```diff
@@ -1486,6 +1486,61 @@ func TestStore_File_Push_DisableOverwrite(t *testing.T) {
...
+		t.Fatal("json.Marshal() error =", err)
+	}
+	manifestDesc := ocispec.Descriptor{
+		MediaType: ocispec.MediaTypeImageManifest,
+		Digest:    digest.FromBytes(manifestJSON),
+		Size:      int64(len(manifestJSON)),
+	}
+	tempDir := t.TempDir()
+	path := filepath.Join(tempDir, "manifest.json")
+	if err := ioutil.WriteFile(path, manifestJSON, 0444); err != nil {
```
Comment (Wwwsylvia) [inline content/file/file_test.go:1512]: Is this file used anywhere?

File: content/file/file_test.go [hunk 3]
```diff
@@ -1486,6 +1486,61 @@ func TestStore_File_Push_DisableOverwrite(t *testing.T) {
...
+	// verify the manifest is not saved
+	exists, err := s.Exists(ctx, manifestDesc)
+	if err != nil {
+		t.Fatal("Store.Exists() error =", err)
+	}
+	if exists {
+		t.Errorf("Unnamed manifest is saved in file store")
+	}
+	// verify the manifest is not indexed
+	predecessors, err := s.Predecessors(ctx, configDesc)
```
Comment (Wwwsylvia) [inline content/file/file_test.go:1535]: It looks like the config itself is not pushed, so it will never have predecessors.

**Review by shizhMSFT — COMMENTED** [review]
General question on how manifest is processed since we need the manifest while the manifests usually does not have a name.

File: content/file/errors.go [hunk 4]
```diff
@@ -20,6 +20,7 @@ import "errors"
 var (
 	ErrMissingName             = errors.New("missing name")
 	ErrDuplicateName           = errors.New("duplicate name")
+	ErrSkipUnnamed             = errors.New("unnamed descriptor skipped")
```
Comment (shizhMSFT) [inline content/file/errors.go:23]: nit: I'm curious that this error is never returned to the user, isn't it?

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**